### ISSUE TITLE
Queries regarding other metrics.

### ISSUE THREAD

**Issue by SSKlearns** [batmanlab/Mammo-CLIP#33]
Dear Authors, 

Thank you so much for releasing your amazing work! I was successfully able to reproduce your results too. However, I attemped to also evaluate the **precision** and **recall** of the released models on the **ViNDr Dataset (calcification and mass classes only)**, and they were pretty low. I understand that this might be a possibility due to the imbalance in the classes, but wanted to compare them with your scores, if you have them. 

Would it be possible to share the precision/recall/f1scores?

Below is ths snippet i added within **inference_loop** of **experiments.py**, drawing inspiration from the rest of the codebase. Kindly do let me know if there is a mismatch between the approach I followed, and your code to calculate these scores.

f1 = f1_score(valid_agg[args.label].values.astype(int), [1 if val > 0.5 else 0 for val in valid_agg['prediction'].values], average='binary')
precision = precision_score(valid_agg[args.label].values.astype(int), [1 if val > 0.5 else 0 for val in valid_agg['prediction'].values])
recall = recall_score(valid_agg[args.label].values.astype(int), [1 if val > 0.5 else 0 for val in valid_agg['prediction'].values])
accuracy = accuracy_score(valid_agg[args.label].values.astype(int), [1 if val > 0.5 else 0 for val in valid_agg['prediction'].values])

Thank you for your assistance!

**Comment by shantanu-ai** [comment]
Hi @SSKlearns 
Thanks for using our work. I did not compute the other metrics as for MICCAI, AUROC is the primary metrics. However, we are extending our work and we are planning to compute other metrics as well. Note, 0.5 is not ideal threshold since the dataset is highly imbalanced. There two good ways of the computing ideal threshold, i am listing two papers. Please check their github as they have computed the thresholds:
1. Algorithmic encoding of protected characteristics in chest X-ray disease detection models. (https://www.thelancet.com/journals/ebiom/article/PIIS2352-3964(23)00032-4/fulltext)
2. The limits of fair medical imaging AI in real-world generalization. (https://www.nature.com/articles/s41591-024-03113-4)

**Comment by SSKlearns** [comment]
Dear Authors,

Thanks for your consistent help across all the issues. Recently, I came across a snippet in the ImageText data class:

```
if self.split == "train":  # Text shuffle augment
    for _text in [text, text2]:
        _text_list = tokenize.sent_tokenize(_text, language="english")
        random.shuffle(_text_list)
        _text = " ".join(_text_list)
```

I understand the intent of this snippet is to augment the text during pretraining, however, python's string immutability will not allow text and text2 to update with this. Sentence swap as an augmentation was mentioned in the paper and I could not find any other references in the code. Kindly let me know in case this is defined elsewhere. In case sentence swap is not taking place in the current code, would you recommend me to fix it and run for a use case (fairly long reports, but the same 10 reports being repeated for a few 100 patients) to prevent the text encoder overfitting, or will this text augmentation not have much impact?

Thank you for your time!

**Comment by shantanu-ai** [comment]
Hey @SSKlearns when we used it, i believe it was working fine. I can check but we created a bigger mammo-clip (named mammo-FM, search it in arxiv paper), so we are moving towards that and gradually this code, we are not maintaining.
